thinc/optimizers.py

```diff
@@ -189,24 +195,35 @@ def __init__(
 
     def _set_attr_or_schedule(self, name, value):
         if isinstance(value, (float, bool, int)):
+            setattr(self, name, constant(value))
+        elif isinstance(value, list):
+            value = iter(value)
+            setattr(self, name, _generator_schedule(name, value))
+        elif isinstance(value, GeneratorType):
+            setattr(self, name, _generator_schedule(name, value))
+        elif isinstance(value, ScheduleCallable):
             setattr(self, name, value)
         else:
-            if isinstance(value, list):
-                value = iter(value)
-            self.schedules[name] = value
-            try:
-                setattr(self, name, next(value))
-            except (StopIteration, TypeError) as e:
-                err = f"Invalid schedule for '{name}' ({type(value)})\n{e}"
-                raise ValueError(err)
+            err = f"Invalid schedule for '{name}' ({type(value)})"
+            raise ValueError(err)
 
     def step_schedules(self):
-        for key, schedule in self.schedules.items():
-            try:
-                value = next(schedule)
-            except StopIteration:  # schedule exhausted, use last value
-                value = getattr(self, key)
-            setattr(self, key, value)
+        self._step += 1
+
+    @property
+    def last_score(self) -> Optional[Tuple[int, float]]:
+        return self._last_score
+
+    @last_score.setter
+    def last_score(self, score: float):
+        self._last_score = (self._step, score)
+
+    def _schedule_args(self, key: KeyT) -> Dict[str, Any]:
```

Comment:
It's not really an `Any` in the return type `Dict[str, Any]`, but I dunno if there is a better way of doing it. It could be some sort of data class maybe like a 

```python
@dataclass
class SchedulerArgs:
   key: Any
   last_score: Comparable
   step: int
```

but maybe its enough like this.

Reply:
I typed it as `Any`, because we may want to add other arguments in the future. We have to make this a dict to do the kwarg expansion (I think?).